homeassistant/components/motionmount/select.py

```diff
@@ -44,11 +44,29 @@ def _update_options(self, presets: list[motionmount.Preset]) -> None:
 
     async def async_update(self) -> None:
         """Get latest state from MotionMount."""
-        presets = await self.mm.get_presets()
-        self._update_options(presets)
-
-        if self._attr_current_option is None:
-            self._attr_current_option = self._attr_options[0]
+        self._presets = await self.mm.get_presets()
+        self._update_options(self._presets)
+
+    @property
+    def current_option(self) -> str | None:
+        """Get the current option."""
+        # When the mount is moving we return the currently selected option
+        if self.mm.is_moving:
+            return self._attr_current_option
+
+        # When the mount isn't moving we select the option that matches the current position
+        self._attr_current_option = None
+        if self.mm.extension == 0 and self.mm.turn == 0:
+            self._attr_current_option = self._attr_options[0]  # Select Wall preset
+        else:
+            for preset in self._presets:
+                if (
+                    preset.extension == self.mm.extension
+                    and preset.turn == self.mm.turn
+                ):
+                    self._attr_current_option = f"{preset.index}: {preset.name}"
+
```

Comment:
Not critical, but I think you can `break` out of the for loop here once you find a match to avoid searching the rest of the list.
```suggestion
                    self._attr_current_option = f"{preset.index}: {preset.name}"
                    break

```